### PR TITLE
reorganize tests, make devdeps testing optional, unpin latest Python testing, and prepare for possibility of data caching

### PR DESCRIPTION
**Pull request by zacharyburnett**
Resolves [SCSB-79](https://jira.stsci.edu/browse/SCSB-79)

This PR reorganizes CI tests and moves devdeps tests to a new optional workflow that can be started with the label `run devdeps tests`

**Checklist**
- [x] ~added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)~
- [x] updated relevant tests
- [x] ~updated relevant documentation~
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: CI
+name: tests
```
Comment (zacharyburnett) [inline .github/workflows/ci.yml:1]:
```suggestion
name: test
```

File: .github/workflows/ci_cron.yml [hunk 2]
```diff
@@ -1,51 +1,57 @@
-
-name: Weekly cron
+name: scheduled tests
```
Comment (zacharyburnett) [inline .github/workflows/ci_cron.yml:1]:
```suggestion
name: test on schedule
```

File: .github/workflows/test_devdeps.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+name: tests with development versions
```
Comment (zacharyburnett) [inline .github/workflows/test_devdeps.yml:1]:
```suggestion
name: test with development versions
```

**Review by hbushouse — APPROVED** [review]

File: .github/workflows/ci_cron.yml [hunk 4]
```diff
@@ -1,51 +1,57 @@
-
-name: Weekly cron
+name: test on schedule
 
 on:
   schedule:
     # Weekly Monday 6AM build
     - cron: "0 0 * * 1"
```
Comment (hbushouse) [inline .github/workflows/ci_cron.yml:6]: Comment line says 6AM, but the use of the "0 0" arguments means midnight (00:00), right?

File: .github/workflows/test_devdeps.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+name: test with development versions
+
+on:
+  push:
+    branches:
+      - master
+      - '*.x'
+    tags:
+      - '*'
+  pull_request:
```
Comment (braingram) [inline .github/workflows/test_devdeps.yml:10]: Should this include "synchronize" "labeled" or some combination of the 2 to allow an existing PR that has the label added to run the workflow (similar to the downstream label and job in asdf: https://github.com/asdf-format/asdf/blob/96800153294b61c271bbf6608ba509cc239dc314/.github/workflows/downstream.yml#L8-L10)?

**Pull request merged**